libsabledb/src/commands/generic_commands.rs

```diff
@@ -170,6 +174,106 @@ impl GenericCommands {
+                            generic_db.put_expiration(key, &exp)?;
+                        }
+                        return Ok(());
+                    }
+                    "xx" => {
+                        // XX -- Set expiry only when the key has an existing expiry
+                        let Some((_, mut value_metadata)) = generic_db.get(key)? else {
+                            return Ok(());
+                        };
+                        if value_metadata.expiration().has_ttl() {
+                            let num = BytesMutUtils::to_u64(seconds);
+                            let mut exp = value_metadata.expiration_mut().clone();
+                            exp.set_expire_timestamp_seconds(num)?;
+                            generic_db.put_expiration(key, &exp)?;
+                        }
+                        return Ok(());
+                    }
+                    "gt" => {
+                        // GT -- Set expiry only when the new expiry is greater than current one
+                        let Some((_, mut value_metadata)) = generic_db.get(key)? else {
+                            return Ok(());
+                        };
+                        if value_metadata.expiration().has_ttl() {
+                            let num = BytesMutUtils::to_u64(seconds);
+                            let mut exp = value_metadata.expiration_mut().clone();
+                            if num > exp.ttl_ms {
+                                exp.set_expire_timestamp_seconds(num)?;
+                                generic_db.put_expiration(key, &exp)?;
+                            }
+                        }
+                    }
+                    "lt" => {
+                        // LT -- Set expiry only when the new expiry is less than current one
+                        let Some((_, mut value_metadata)) = generic_db.get(key)? else {
+                            return Ok(());
+                        };
+                        if value_metadata.expiration().has_ttl() {
+                            let num = BytesMutUtils::to_u64(seconds);
+                            let mut exp = value_metadata.expiration_mut().clone();
+                            if num < exp.ttl_ms {
+                                exp.set_expire_timestamp_seconds(num)?;
+                                generic_db.put_expiration(key, &exp)?;
+                            }
+                        }
+                    }
+                    _ => {
+                        // prepare error message to the user
+                        builder.error_string(response_buffer, "...");
+                        return Ok(());
+                    }
+                }
+            }
+            (Some(key), Some(seconds), None) => {
+                if let Some(expiration) = generic_db.get_expiration(key)? {
+                    let mut exp = expiration.clone();
+                    let num = BytesMutUtils::to_u64(seconds);
```

Comment:
This could be simplified like this:

```rust
    let mut expiration = match generic_db.get_expiration(key)? {
        Some(expiration) => expiration,
        // since we override, create a default expiration and set its value
        None => Expiration::default(),
    };
```

In addition, we could remove the duplicate code and load the expiration at the top